[Bug]: Build fails with Go 1.23 because of changed testing.testDeps

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
If the matter is security sensitive, please disclose it privately to a security contact: https://github.com/kudobuilder/kuttl/blob/master/SECURITY_CONTACTS
-->


**What happened**:

build fails with 
````
# github.com/kudobuilder/kuttl/pkg/test/utils
  pkg/test/utils/testing.go:44:28: cannot use &testDeps{} (value of type *testDeps) as "testing".testDeps value in argument to testing.MainStart: *testDeps does not implement "testing".testDeps (missing method InitRuntimeCoverage)
````

**What you expected to happen**:

build to succeed.

**How to reproduce it (as minimally and precisely as possible)**:
1) Install Go 1.23 (rc1, rc2 or later)
2) build

**Anything else we need to know?**:

Related:
* found while building with https://github.com/Homebrew/homebrew-core/pull/175310
* similar issue: https://github.com/hofstadter-io/hof/issues/391

**Environment**:
- Kubernetes version (use `kubectl version`):
- KUTTL version (use `kubectl kuttl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/pkg/test/utils/testing.go
+++ b/pkg/test/utils/testing.go
@@ -116,6 +116,10 @@ func (testDeps) SnapshotCoverage() {
 
 }
 
+func (testDeps) InitRuntimeCoverage() (mode string, tearDown func(string, string) (string, error), snapcov func() float64) {
+	return
+}
+
 // corpusEntry is from the public go testing which references an internal structure.
 // corpusEntry is an alias to the same type as internal/fuzz.CorpusEntry.
 // We use a type alias because we don't want to export this type, and we can't

